File: Code/new_fitervari_backend/src/main/java/at/htl/model/Workout.java
package at.htl.model;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "Workout")
public class Workout {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    @Column(name = "creation_Date")
    private LocalDateTime creationDate;

    @ManyToOne(cascade = CascadeType.MERGE, fetch = FetchType.EAGER)
    private Person creator;

    @Column(name = "official_flag")
    private boolean officialFlag;

    @ManyToMany(cascade = CascadeType.MERGE, fetch = FetchType.EAGER)
    @JoinTable(
            name = "workout2exercise",
            joinColumns = @JoinColumn(name = "workout_id"),
            inverseJoinColumns = @JoinColumn(name = "exercise_id")
    )
    private List<Exercise> exercises;

    @OneToMany(cascade = CascadeType.MERGE)
    @JoinColumn(name = "workout_id")
    private List<WorkoutHistory> workoutHistories;

    //region GetterSetter
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public LocalDateTime getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(LocalDateTime creationDate) {
        this.creationDate = creationDate;
    }

    public Person getCreator() {
        return creator;
    }

    public void setCreator(Person creator) {
        this.creator = creator;
    }

    public boolean isOfficialFlag() {
        return officialFlag;
    }

    public void setOfficialFlag(boolean officialFlag) {
        this.officialFlag = officialFlag;
    }

    public List<Exercise> getExercises() {
        return exercises;
    }

    public void setExercises(List<Exercise> exercises) {
        this.exercises = exercises;
    }

    public List<WorkoutHistory> getWorkoutHistories() {
        return workoutHistories;
    }

    public void setWorkoutHistories(List<WorkoutHistory> workoutHistories) {
        this.workoutHistories = workoutHistories;
    }
    //endregion
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/model/Workout.kt
package at.htl_leonding.model

import java.time.LocalDateTime
import javax.persistence.*


@Entity
@Table(name = "Workout")
class Workout(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    var name: String,
    var creation_Date: LocalDateTime,
    @ManyToOne(cascade = [CascadeType.MERGE])
    var creator: Person,
    var official_Flag: Boolean,
    @ManyToMany(cascade = [CascadeType.MERGE])
    @JoinTable(
        name = "workout2exercise",
        joinColumns = [JoinColumn(name = "workout_id")],
        inverseJoinColumns = [JoinColumn(name = "exercise_id")]
    )
    var exercises: MutableList<Exercise> = mutableListOf(),
    @OneToMany(cascade = [CascadeType.MERGE])
    @JoinColumn(name = "workout_id")
    var workouthistories: MutableList<WorkoutHistory> = mutableListOf()
) {
    fun copyValues(other: Workout) {
        this.name = other.name
        this.creation_Date = other.creation_Date
        this.creator = other.creator
        this.official_Flag = other.official_Flag
    }
}